src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -0,0 +1,480 @@
+            sync_properties=SyncProperties(
+                token_id=sync_token_id,
+                schedule=sync_schedule,
+                message_ttl=sync_message_ttl,
+                sync_window=sync_window
+            )
+        ),
+        client_token_ids=client_token_ids,
+        logging=LoggingProperties(
+            log_level=log_level,
+            audit_log_status='Enabled' if sync_audit_logs_enabled else 'Disabled'
+        )
+    )
+
+    try:
+        return client.create(subscription_id=subscription_id,
+                             resource_group_name=resource_group_name,
+                             registry_name=registry_name,
+                             connected_registry_name=connected_registry_name,
+                             connected_registry_create_parameters=connected_registry_create_parameters)
+    except Exception as e:
+        # Delete the previously created sync scope map. 
+        #scope_map_client = cf_acr_scope_maps(cmd.cli_ctx)
+        #scope_map_client.delete(resource_group_name, registry_name, SYNC_SCOPE_MAP_NAME.format(connected_registry_name))
+        raise CLIError(e)
+
+
+def acr_connected_registry_update(cmd,
+                                  client,
+                                  registry_name,
+                                  connected_registry_name,
+                                  add_client_token_ids=None,
+                                  remove_client_token_ids=None,
+                                  resource_group_name=None,
+                                  sync_schedule=None,
+                                  sync_window=None,
+                                  log_level=None,
+                                  sync_message_ttl=None,
+                                  sync_audit_logs_enabled=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    current_connected_registry = acr_connected_registry_show(
+        cmd, client, connected_registry_name, registry_name, resource_group_name)
+
+    # Add or remove from the current client token id list
+    add_client_token_set = set(add_client_token_ids) if add_client_token_ids else set()
+    remove_client_token_set = set(remove_client_token_ids) if remove_client_token_ids else set()
+    duplicate_client_token = set.intersection(add_client_token_set, remove_client_token_set)
+    if duplicate_client_token:
+        errors = sorted(map(lambda action: action[action.find('/') + 1:], duplicate_client_token))
+        raise CLIError(
+            'Update ambiguity. Duplicate client token ids were provided with ' +
+            '--add-client-token-ids and --remove-client-token-ids arguments.\n{}'.format(errors))
+
+    if sync_message_ttl is None:
+        sync_message_ttl = current_connected_registry.parent.sync_properties.message_ttl
+    ConnectedRegistryUpdateParameters, SyncProperties, LoggingProperties = cmd.get_models(
```

Comment:
@rosanch please import `SyncUpdateProperties` model for PATCH instead of `SyncProperties`. `SyncProperties` is used for PUT.